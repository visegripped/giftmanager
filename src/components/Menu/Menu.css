.menu-item a {
  text-decoration: none;
}