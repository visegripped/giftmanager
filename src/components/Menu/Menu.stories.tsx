import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import Menu from './index';

export default {
  title: 'component/Menu',
  component: Menu,
} as ComponentMeta<typeof Menu>;

const Template: ComponentStory<typeof Menu> = (args) => (
  <BrowserRouter>
    <Menu {...args} />
  </BrowserRouter>
);

export const BasicUsage = Template.bind({});
BasicUsage.args = {
  items: [
    {
      link: '#nowhere',
      value: 'something',
    },
    {
      link: '#nowhere',
      value: 'something else',
    },
  ],
};
