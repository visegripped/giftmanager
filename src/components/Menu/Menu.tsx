import React from 'react';
import { Link } from 'react-router-dom';
import './Menu.css';

export interface MenuItemProps {
  link: string;
  value: string;
}

interface MenuProps {
  items: MenuItemProps[];
}

const Menu = (props: MenuProps) => {
  const { items } = props;
  return (
    <menu data-testid="Menu">
      {items.map((item: MenuItemProps) => {
        const { link, value } = item;
        return (
          <li className="menu-item">
            <Link to={link}>{value}</Link>
          </li>
        );
      })}
    </menu>
  );
};

export default Menu;
