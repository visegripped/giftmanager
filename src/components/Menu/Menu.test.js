import { render, screen } from '@testing-library/react';
import Menu from './index';
import { BrowserRouter } from 'react-router-dom';

describe('Menu', () => {
  test('snapshot matches', () => {
    render(<Menu cssClasses="test classes" />, { wrapper: BrowserRouter });
    const renderedMenu = screen.getByTestId('Menu');
    expect(renderedMenu).toMatchSnapshot();
  });
});
