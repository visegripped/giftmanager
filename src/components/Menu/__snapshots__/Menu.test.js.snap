// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot matches 1`] = `
<menu
  data-testid="Menu"
/>
`;
